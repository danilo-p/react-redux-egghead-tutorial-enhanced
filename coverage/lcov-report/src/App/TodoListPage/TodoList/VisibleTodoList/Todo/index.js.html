<!doctype html>
<html lang="en">
<head>
    <title>Code coverage report for src/App/TodoListPage/TodoList/VisibleTodoList/Todo/index.js</title>
    <meta charset="utf-8" />
    <link rel="stylesheet" href="../../../../../../prettify.css" />
    <link rel="stylesheet" href="../../../../../../base.css" />
    <meta name="viewport" content="width=device-width, initial-scale=1">
    <style type='text/css'>
        .coverage-summary .sorter {
            background-image: url(../../../../../../sort-arrow-sprite.png);
        }
    </style>
</head>
<body>
<div class='wrapper'>
  <div class='pad1'>
    <h1>
      <a href="../../../../../../index.html">All files</a> / <a href="index.html">src/App/TodoListPage/TodoList/VisibleTodoList/Todo</a> index.js
    </h1>
    <div class='clearfix'>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Statements</span>
        <span class='fraction'>3/3</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Branches</span>
        <span class='fraction'>0/0</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Functions</span>
        <span class='fraction'>1/1</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Lines</span>
        <span class='fraction'>3/3</span>
      </div>
    </div>
  </div>
  <div class='status-line high'></div>
<pre><table class="coverage">
<tr><td class="line-count quiet">1
2
3
4
5
6
7
8
9
10
11
12
13
14
15
16
17
18
19
20
21
22
23
24
25
26
27
28
29
30
31
32
33
34
35
36
37</td><td class="line-coverage quiet"><span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">6x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">44x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">6x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span></td><td class="text"><pre class="prettyprint lang-js">import React from 'react';
import PropTypes from 'prop-types';
import { Checkbox } from 'semantic-ui-react';
import styles from './index.styl';
&nbsp;
/**
 * Describes a todo item.
 *
 * @param {Object} props The Todo's component properties
 * @param {Object} props.onClick The todo's click handler
 * @param {Object} props.completed The todo's completeness
 * @param {Object} props.text The todo's content
 * @returns {jsx} The Todo's component template
 */
const Todo = ({
  onClick,
  completed,
  text,
}) =&gt; (
  &lt;li&gt;
    &lt;Checkbox
      className={styles.Todo__Checkbox}
      onClick={onClick}
      label={{ children: text }}
      checked={completed}
    /&gt;
  &lt;/li&gt;
);
&nbsp;
Todo.propTypes = {
  onClick: PropTypes.func.isRequired,
  completed: PropTypes.bool.isRequired,
  text: PropTypes.string.isRequired,
};
&nbsp;
export default Todo;
&nbsp;</pre></td></tr>
</table></pre>
<div class='push'></div><!-- for sticky footer -->
</div><!-- /wrapper -->
<div class='footer quiet pad2 space-top1 center small'>
  Code coverage
  generated by <a href="https://istanbul.js.org/" target="_blank">istanbul</a> at Thu Jan 04 2018 00:21:52 GMT-0200 (-02)
</div>
</div>
<script src="../../../../../../prettify.js"></script>
<script>
window.onload = function () {
        if (typeof prettyPrint === 'function') {
            prettyPrint();
        }
};
</script>
<script src="../../../../../../sorter.js"></script>
</body>
</html>
