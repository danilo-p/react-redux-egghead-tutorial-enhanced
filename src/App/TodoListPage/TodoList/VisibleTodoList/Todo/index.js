import React from 'react';
import PropTypes from 'prop-types';
import { Checkbox } from 'semantic-ui-react';
import styles from './index.styl';

/**
 * Describes a todo item.
 *
 * @param {Object} props The Todo's component properties
 * @param {Object} props.onClick The todo's click handler
 * @param {Object} props.completed The todo's completeness
 * @param {Object} props.text The todo's content
 * @returns {jsx} The Todo's component template
 */
const Todo = ({
  onClick,
  completed,
  text,
}) => (
  <li>
    <Checkbox
      className={styles.Todo__Checkbox}
      onClick={onClick}
      label={{ children: text }}
      checked={completed}
    />
  </li>
);

Todo.propTypes = {
  onClick: PropTypes.func.isRequired,
  completed: PropTypes.bool.isRequired,
  text: PropTypes.string.isRequired,
};

export default Todo;
