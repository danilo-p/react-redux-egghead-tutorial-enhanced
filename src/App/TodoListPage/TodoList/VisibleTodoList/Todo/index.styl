.Todo__Checkbox:global(.checked) label
  text-decoration: line-through;