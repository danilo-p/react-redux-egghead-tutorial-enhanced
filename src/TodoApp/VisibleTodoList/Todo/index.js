import React from 'react';
import PropTypes from 'prop-types';
import { Checkbox } from 'semantic-ui-react';

import styles from './index.styl';

const Todo = ({
  onClick,
  completed,
  text,
}) => (
  <li>
    <Checkbox
      className={styles.Todo__Checkbox}
      onClick={onClick}
      label={{ children: text }}
      checked={completed}
    />
  </li>
);

Todo.propTypes = {
  onClick: PropTypes.func.isRequired,
  completed: PropTypes.bool.isRequired,
  text: PropTypes.string.isRequired,
};

export default Todo;
