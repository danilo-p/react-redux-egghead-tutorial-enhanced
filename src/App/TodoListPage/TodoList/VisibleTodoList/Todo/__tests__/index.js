import React from 'react';
import basicComponentTests from 'tests/basicComponentTests';
import Todo from '../';

describe('Todo', () => {
  const component = (
    <Todo
      completed={false}
      text="test"
      onClick={jest.fn()}
    />
  );
  basicComponentTests(component);
});
