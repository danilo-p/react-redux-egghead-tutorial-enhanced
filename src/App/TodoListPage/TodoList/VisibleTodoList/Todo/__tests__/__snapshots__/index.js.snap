// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Todo should render the component 1`] = `
<li>
  <Checkbox
    checked={false}
    className="Todo__Checkbox"
    label={
      Object {
        "children": "test",
      }
    }
    onClick={[Function]}
    type="checkbox"
  />
</li>
`;
